packages/app-cli/app/gui/FolderListWidget.ts

```diff
@@ -4,12 +4,97 @@ import BaseModel from '@joplin/lib/BaseModel';
 import Setting from '@joplin/lib/models/Setting';
 import { _ } from '@joplin/lib/locale';
 import { FolderEntity } from '@joplin/lib/services/database/types';
-import { getDisplayParentId, getTrashFolderId } from '@joplin/lib/services/trash';
+import {
+	getDisplayParentId,
+	getTrashFolderId,
+} from '@joplin/lib/services/trash';
 const ListWidget = require('tkwidgets/ListWidget.js');
 
-export default class FolderListWidget extends ListWidget {
+// Service for managing folder collapse states
+class FolderCollapseService {
```

Comment:
What is the need for this service? Presumably it shouldn't need to manage any state since everything is saved/loaded from the settings? Settings already handle caching so it's fast to read/write from them